### PR TITLE
Add previewLiquidate/Liquidate functionality

### PR DESCRIPTION
**Pull request by torreyatcitty**
- Added in liquidation selectors and input amount so users can specify how they want to liquidate an underwater borrow.
- Added in working preview liquidate button and UI to show user's how much collateral they might receive.
- Added in working liquidate button.

### CODE REVIEW

**Comment by hayesgm** [comment]
Thanks for the comment. Standing by!

**Comment by torreyatcitty** [comment]
ok thanks for standing by. All ready now @hayesgm

File: src/elm/Liquidate.elm [hunk 1]
```diff
@@ -75,6 +88,13 @@ type InternalMsg
     = NewLedgerEntries (List LedgerEntry)
     | CollateralShortfallsUpdate (List CollateralShortfall)
     | SelectBorrow BorrowerDetail
+    | SetBorrowedAsset String
+    | SetDesiredCollateral String
+    | LiquidateAmountInputChanged String
+    | AskPreviewLiquidate TokenSet LiquidateBorrowDetails
```
Comment (hayesgm) [inline src/elm/Liquidate.elm]: `LiquidateBorrowDetails` gets a lot of use!

File: src/elm/Liquidate.elm [hunk 2]
```diff
@@ -166,6 +274,56 @@ update msg model contract =
...
+        AskPreviewLiquidate tokens liquidateBorrowDetails ->
+            ( model, buildLiquidateRequestCmd account tokens model.maybeMoneyMarket liquidateBorrowDetails msg )
+
+        -- Preview Liquidate Update is an asynchronous response so we need
+        -- to verify the update is for the currently selected borrow and
+        -- preview liquidate parameters.
+        PreviewLiquidateResponse liquidateBorrowDetails ->
+            let
+                updatedModel =
+                    case (model.selectedBorrow) of
```
Comment (hayesgm) [inline src/elm/Liquidate.elm]: Probably can remove the parens.

File: src/elm/Liquidate.elm [hunk 3]
```diff
@@ -166,6 +274,56 @@ update msg model contract =
             in
                 ( { model | selectedBorrow = Just borrowerDetail }, loadBalancesCmd )
 
+        SetBorrowedAsset assetAddress ->
+            ( handleSelectedBorrowUpdate model msg, Cmd.none )
```
Comment (hayesgm) [inline src/elm/Liquidate.elm]: I like this strategy. Looks pretty suave.

File: src/elm/Liquidate.elm (hunk 2 above)
Comment (torreyatcitty) [reply src/elm/Liquidate.elm]: yup fixed.

**Pull request merged**